tools/wptrunner/wptrunner/executors/executorchrome.py

```diff
@@ -1,9 +1,122 @@
+import os
+import traceback
+
+from six.moves.urllib.parse import urljoin
+
+from .base import WdspecExecutor, get_pages
+from .executorwebdriver import WebDriverProtocol, WebDriverRefTestExecutor, WebDriverRun
+from .protocol import PrintProtocolPart
 from ..webdriver_server import ChromeDriverServer
-from .base import WdspecExecutor, WebDriverProtocol
+
+here = os.path.join(os.path.split(__file__)[0])
+
+
+class ChromeDriverPrintProtocolPart(PrintProtocolPart):
+    def setup(self):
+        self.webdriver = self.parent.webdriver
+        self.runner_handle = None
+
+    def load_runner(self):
+        url = urljoin(self.parent.executor.server_url("http"), "/print_reftest_runner.html")
+        self.logger.debug("Loading %s" % url)
+        try:
+            self.webdriver.url = url
+        except Exception as e:
+            self.logger.critical(
+                "Loading initial page %s failed. Ensure that the "
+                "there are no other programs bound to this port and "
+                "that your firewall rules or network setup does not "
+                r"prevent access.\e%s" % (url, traceback.format_exc(e)))
+            raise
+        self.runner_handle = self.webdriver.window_handle
+
+    def render_as_pdf(self, width, height):
+        page_ranges = self.webdriver.execute_script(
+            """
+let elem = document.querySelector("meta[name=reftest-pages]");
+return elem ? elem.content : null;
+""")
+        margin = 0.5
+        body = {
+            "cmd": "Page.printToPDF",
+            "params": {
+                "paperWidth": width / 2.54,
```

Comment:
I assume then that the chrome-specific endpoint here only accepts inches, not cm? Worth documenting that if so, otherwise the `2.54` is sorta magical.